README.md

```diff
@@ -1,2 +1,77 @@
 # artifact-upload-handler
+
 A secure webserver to handle uploading CI conda build artifacts to a specified conda channel.
+
+To use, simply clone this repo and run the Python webserver, observing the
+[Usage](#usage) and [Requirements](#requirements) instructions below.
+
+
+## Usage
+
+```
+usage: artifact_upload_handler.py [-h] -d WRITE_DIR -p PORT -e CONDA_EXE -c
+                                  CERTFILE -k KEYFILE -t TOKEN_HASH
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -d WRITE_DIR, --write_dir WRITE_DIR
+                        directory to write artifacts to
+  -p PORT, --port PORT  webserver port number
+  -e CONDA_EXE, --conda_exe CONDA_EXE
+                        full path to conda executable
+  -c CERTFILE, --certfile CERTFILE
+                        full path to certificate file for SSL
+  -k KEYFILE, --keyfile KEYFILE
+                        full path to keyfile for SSL
+  -t TOKEN_HASH, --token_hash TOKEN_HASH
+                        secure token hash
+```
+
+
+### Usage Example 
+
+For example, to start a secure webserver running on port 9999 that will write
+to the (imaginary) conda channel at ``/path/to/conda/channel``:
+
+```
+$ python artifact_upload_handler.py -d /path/to/conda/channel \
+                                    -p 9999 \
+                                    -e /path/to/env/bin/conda \
+                                    -c /path/to/mycertfile.crt \
+                                    -k /path/to/mykeyfile.key \
+                                    -t eccd989b
+
+```
+
+To test the webserver:
+
+```python
+import requests
+
+artifacts = [open('my-artifact1-1.0.0-2.tar.bz2', 'rb'),
+             open('my-artifact2-3.4.2-0.tar.bz2', 'rb'),
+             open('my-artifact3-0.9.1-0.tar.bz2', 'rb'),
+            ]
+url = 'https://localhost:9999/'
+token = 'mysecuretoken'
+
+for artifact in artifacts:
+    requests.post(url, data={'token': token}, files={'artifact': artifact},
+                  verify=False)
```

Comment:
@dkillick Is there a need to explain the usage of `verify` ?

Reply:
That's what the requests docs are for.

Reply:
Fair point, but a link to [SSL Cert Verification](http://docs.python-requests.org/en/master/user/advanced/#ssl-cert-verification) would be kinda nice, right?